File: csharp/calculator-conundrum/CalculatorConundrum.cs
using System;

public static class SimpleCalculator
{
    public static string Calculate(int operand1, int operand2, string operation)
    {
        if (operation == "+")
        {
            var sum = operand1 + operand2;
            return $"{operand1} + {operand2} = {sum}";
        }
        else if (operation == "*")
        {
            var product = operand1 * operand2;
            return $"{operand1} * {operand2} = {product}";
        }
        else if (operation == "/")
        {
            float quotient;
            try
            {
                quotient = operand1 / operand2;
            }
            catch (DivideByZeroException)
            {
                return "Division by zero is not allowed.";
            }
            return $"{operand1} / {operand2} = {quotient}";
        }
        else
        {
            if (operation == string.Empty)
            {
                throw new ArgumentException("Operation cannot be empty");
            }
            else if(operation == null)
            {
                throw new ArgumentNullException("Operation cannot be null");
            }
            else
            {
                throw new ArgumentOutOfRangeException($"Operation {operation} is not supported");
            }
        }
    }
}

File: java/calculator-conundrum/src/main/java/CalculatorConundrum.java
class CalculatorConundrum {
    public String calculate(int operand1, int operand2, String operation) {
        switch (operation) {
            case null:
                throw new IllegalArgumentException("Operation cannot be null");
            case "":
                throw new IllegalArgumentException("Operation cannot be empty");
            case "+":
                return String.format("%d + %d = %d", operand1, operand2, operand1 + operand2);
            case "*":
                return String.format("%d * %d = %d", operand1, operand2, operand1 * operand2);
            case "/":
                try {
                    return String.format("%d / %d = %d", operand1, operand2, operand1 / operand2);
                } catch (ArithmeticException e) {
                    throw new IllegalOperationException("Division by zero is not allowed", e);
                }

            default:
                throw new IllegalOperationException("Operation '" + operation + "' does not exist");
        }
    }
}
